### DISCUSSION TITLE
Using filename in the rewrite expression?

### DISCUSSION THREAD

**Discussion by Smibu** [ast-grep/ast-grep#764] [Q&A]
Is there a way use the current filename in the rewrite expression?

Use case: I'd like to replace some default exports with named exports (in a TS codebase), with the name coming from filename.

For example, when processing file `/path/to/somefile.ts` it would give `somefile`.

**Comment by HerringtonDarkholme** [comment]
Hi, this is not possible using rule/cli at the moment.

https://ast-grep.github.io/guide/api-usage/js-api.html is a good solution at the moment.
